src/index.js

```diff
@@ -4,58 +4,51 @@ import './index.css';
 
 function Square(props) {
 
-  if (props.value === 'X') {
     return (
       <button 
-        className={"square X " + (props.Winning ? "victory-square" : null)} 
+        className={"square " + (props.Winning ? "victory-square " : null) + (props.value === 'X' ? " X" : " O")} 
         onClick={props.onClick}
       >
         {props.value}
       </button>
     );
-  } else {
-    return (
-      <button 
-        className={"square O " + (props.Winning ? "victory-square" : null)} 
-        onClick={props.onClick}
-      >
-        {props.value}
-      </button>
-    );
-  }
 }
 
 //Board
 
 class Board extends React.Component {
   
-  renderSquare(i) {
-    return (<Square 
-      Winning = {this.props.winningSquares.includes(i)}
-	    value={this.props.squares[i]} 
-	    onClick={() => this.props.onClick(i)} 
-	  />
-    );
+  makeRow(x) {
+    return <div className="board-row">{x}</div>;
   }
 
+  makeBoard(n) {
+    var board = []
+    for (let i = 0; i < n; i++) {
+      let row = [];
+      for (let j = 0; j < n; j++) {
+        row.push(i*n + j);
+      }
+      board.push(row);
+    }
+    return board;
+  }
+  
   render() {
+    var board = this.makeBoard(this.props.boardSize);
     return (
       <div>
-        <div className="board-row">
-          {this.renderSquare(0)}
-          {this.renderSquare(1)}
-          {this.renderSquare(2)}
-        </div>
-        <div className="board-row">
-          {this.renderSquare(3)}
-          {this.renderSquare(4)}
-          {this.renderSquare(5)}
-        </div>
-        <div className="board-row">
-          {this.renderSquare(6)}
-          {this.renderSquare(7)}
-          {this.renderSquare(8)}
-        </div>
+        {board.map(
```

Comment:
You probably don't need method `makeRow`

```
{board.map((row) =>
    <div className="board-row">
    {row.map((square) =>
       <Square 
          Winning = {this.props.winningSquares.includes(x)}
          value={this.props.squares[square]} 
          onClick={() => this.props.onClick(square)} 
       />
     )}
  </div>
)}
```

Reply:
Good suggestion, fixed it

`Winning = {this.props.winningSquares.includes(x)}` should be `Winning = {this.props.winningSquares.includes(square)}`